image/png.ksy

```diff
@@ -292,6 +305,68 @@ types:
       - id: text_datastream
         process: zlib
         size-eos: true
+  animation_control_chunk:
+    doc-ref: https://wiki.mozilla.org/APNG_Specification#.60acTL.60:_The_Animation_Control_Chunk
+    seq:
+      - id: num_frames
+        type: u4
+        doc: Number of frames
+      - id: num_plays
+        type: u4
+        doc: Number of times to loop, 0 indicates infinite looping.
+  frame_control_chunk:
+    doc-ref: https://wiki.mozilla.org/APNG_Specification#.60fcTL.60:_The_Frame_Control_Chunk
+    seq:
+      - id: sequence_number
+        type: u4
+        doc: Sequence number of the animation chunk
+      - id: width
+        type: u4
+        valid:
+          max: _root.ihdr.width
+        doc: Width of the following frame
+      - id: height
+        type: u4
+        valid:
+          max: _root.ihdr.height
+        doc: Height of the following frame
+      - id: x_offset
+        type: u4
+        valid:
+          max: _root.ihdr.width - width
+        doc: X position at which to render the following frame
+      - id: y_offset
+        type: u4
+        valid:
+          max: _root.ihdr.height - height
+        doc: Y position at which to render the following frame
+      - id: delay_num
+        type: u2
+        doc: Frame delay fraction numerator
+      - id: delay_den
+        type: u2
+        doc: Frame delay fraction denominator
+      - id: dispose_op
+        type: u1
+        enum: dispose_op_values
+        doc: Type of frame area disposal to be done after rendering this frame
+      - id: blend_op
+        type: u1
+        enum: blend_op_values
+        doc: Type of frame area rendering for this frame
+    instances:
+      delay:
+        value: 'delay_num / (delay_den == 0 ? 100.0 : delay_den)'
+        doc: delay in seconds
```

Comment:
Probably it makes sense to write this a little bit more descriptively. The [spec](https://wiki.mozilla.org/APNG_Specification#.60fcTL.60:_The_Frame_Control_Chunk) says

> The `delay_num` and `delay_den` parameters together specify a fraction **indicating the time to display the current frame, in seconds**.

and I suggest

```suggestion
        doc: Time to display this frame, in seconds
```